### PR TITLE
Address all reflection warnings

### PR DESCRIPTION
**Pull request by vemv**
> Closes https://github.com/clj-commons/pomegranate/pull/124

Picks up https://github.com/clj-commons/pomegranate/pull/124 from @emlyn. It's generally a fine commit except for Clojure 1.7 compatibility, and for some further opportunity of addressing the remaining reflection warnings.

That is done in the two following commits.

And the last commit sets up Eastwood as a linter. Starting from 0.9, it lints for reflection warnings in a way that is suitable for CI. You can learn more about how it works / its rationale in https://github.com/jonase/eastwood#reflection

### CODE REVIEW

**Comment by vemv** [comment]
As some context, Pomegranate's reflection warnings were being a nuisance here https://github.com/clojure-emacs/refactor-nrepl/pull/301#issuecomment-868502330

File: src/main/clojure/cemerick/pomegranate/aether.clj [hunk 1]
```diff
@@ -170,22 +170,22 @@
 
 (defn- set-policies
   [repo-builder settings]
-  (doto repo-builder
+  (doto ^org.eclipse.aether.repository.RemoteRepository$Builder repo-builder
```
Comment (slipset) [inline src/main/clojure/cemerick/pomegranate/aether.clj:173]: Import?

File: src/main/clojure/cemerick/pomegranate/aether.clj [hunk 2]
```diff
@@ -196,13 +196,14 @@
...
-          prx (.getProxy prx-sel (.build repo-builder))] ; ugg.
+          prx (.getProxy prx-sel (.build ^org.eclipse.aether.repository.RemoteRepository$Builder repo-builder))] ; ugg.
       ;; Don't know how to get around "building" the repo for this
-      (.setProxy repo-builder prx))
+      (.setProxy ^org.eclipse.aether.repository.RemoteRepository$Builder repo-builder prx))
     repo-builder))
 
 (defn make-repository
   "Produces an Aether RemoteRepository instance from Pomegranate-style repository information"
+  ^org.eclipse.aether.repository.RemoteRepository
```
Comment (slipset) [inline src/main/clojure/cemerick/pomegranate/aether.clj:206]: Import?

**Review by slipset — APPROVED** [review]

File: src/main/clojure/cemerick/pomegranate/aether.clj (hunk 1 above)
Comment (vemv) [reply src/main/clojure/cemerick/pomegranate/aether.clj:173]: `^RemoteRepository$Builder` would fail for clojure 1.7

**Pull request merged**